### ISSUE TITLE
Docker image

### ISSUE THREAD

**Issue by newkind** [rileytestut/AltStore#42]
I'd like to request a possibility of using AltStore as a docker image. 

Why a docker image?

Well just imagine that you install this image on your server or NAS and then, when you plan to travel for more than a week (and you don't have a laptop with you), all you have to do is to connect to your home network via VPN so that both devices are on the same network and all of the AltStore apps could get auto-refreshed :)

We could use the docker environment variables to store the AppleID username/password/app-password for the ease of configurability.

**Comment by TheManchineel** [comment]
1. I don't think you can use iTunes Wi-Fi sync over VPN
2. AltStore would have to be available on Linux first.

**Comment by StiviiK** [comment]
>I don't think you can use iTunes Wi-Fi sync over VPN
https://github.com/m4rkw/iTunesRemoteSync this is a very old project but it looks like there was the idea to tackle this issue.

**Comment by TheManchineel** [comment]
@StiviiK the repo you linked contains a piece of software that relies on a computer (namely a Mac) running a proxy while connected to the iPhone via Wi-Fi, in addition to the remote Mac. iTunes Wi-Fi Sync uses multicast DNS, so it can't be done with a VPN or just the iDevice alone. You might as well be running the server on the Mac itself by that point.

**Comment by StiviiK** [comment]
Yeah, that was what I meant with limited functionality.

**Comment by LucasMNP** [comment]
I was searching about it this week and I found this open issue.
It'd be great to have AltStore as a Docker container. It doesn't even needs to work via VPN for me, running it on my NAS and it working while I'm home would already be a big advantage, since I don't turn my computer on always and as it's in my bedroom I don't keep it powered on during the night. It'd be nice to keep AltStore always running on the container so it can resing the apps during the night and I wouldn't have to remember to turn my computer on at least once a week or turn the computer on every time I want to install an app.
I hope this turns real some day!

**Comment by LucasMNP** [comment]
This is probably possible now that you can run macOS on docker (see https://github.com/sickcodes/Docker-OSX)

**Comment by TheManchineel** [comment]
@LucasMNP ![image](https://user-images.githubusercontent.com/37479927/84034146-3b1c7580-a99a-11ea-8c0d-8704bf002502.png)

**Comment by TheManchineel** [comment]
It seems to rely on availability of KVM on the host. Could probably run on Unraid but performance might be abysmal. I'll give this one a try and see how well it works.

**Comment by StiviiK** [comment]
https://github.com/sickcodes/Docker-OSX

**Comment by taw123** [comment]
Performance would be irrelevant (and running an entire virtualized instance of Mac OS just for AltStore IS HUGELY overkill)..... 

However if you tie it into a Shortcut one might create running in iOS that would send a request to say my QNAP when iOS location == home that would start the Mac/AltStore docker VM. Combined with a cron job running on the NAS (not in MAC vm) that after would ssh into the Mac VM to perform orderly shutdown (or set up cron job running in the VM to shutdown after say 2 hours assuming AltStore will have noticed the server is up and refreshed signing), then finally the script would stop the Docker container...  

There you would really have something...   IF AltStore does run in the VM, and performance is tolerable (when restricted to not impact other processes on my NAS), then this shouldn't be to hard for me to put together....   And would add to the list with trying to get Checkm8 running on the NAS as well 😉

**Comment by superkrups20056** [comment]
Any updates on this awesome idea?

**Comment by DanTheMan827** [comment]
With there now being a windows version, it might be possible to port it over to Linux as well given that the code is C++

**Comment by xXBannedXx** [comment]
> With there now being a windows version, it might be possible to port it over to Linux as well given that the code is C++

No, since alt store depends on iCloud, and there is no official support for iCloud on Linux

**Comment by TheManchineel** [comment]
Probably a stupid question, but how does WineHQ handle iCloud at the moment?

**Comment by superkrups20056** [comment]
It is available on Linux now. Any way we can get it working for Synology? https://www.idownloadblog.com/2022/04/03/altserver-linux-gui/

**Comment by superkrups20056** [comment]
Chiming in after year three on this thread. Any updates?
